File: Đề NĐ các năm/20-21/THPT GIAO THUY/I67201/ghh.pas
var fa,fb:text;
    n,i,j,dem:longint;
    a:array[0..10005] of longint;
    function tu(x:longint):longint;
    var
        t,s:longint;
    begin
      s:=0;
      for t:=1 to trunc(Sqrt(x)) do
        begin
          if x mod t = 0 then
                s:=s+t+ x div t;
        end;
      if trunc(sqrt(x))=sqrt(X) then s:=s-trunc(Sqrt(x));
      exit(s);
    end;
begin
    assign(fa,'ghh.inp');reset(fa);
    assign(fb,'ghh.out');rewrite(fb);
    readln(fa,n);
    for i:=1 to n do
      begin
        readln(Fa,j);
        if 2*j<=tu(j) then begin inc(dem);a[dem]:=j; end;
      end;
    writeln(fb,dem);
    for i:=1 to dem do
       writeln(fb,a[i]);
close(fa);
close(fb);
end.

File: Đề NĐ các năm/20-21/Bai XTB 2021/I91201/GHH.cpp
#include <iostream>
#include<fstream>
using namespace std;
bool check (int n);
int main()
{
    int n;
    int dem =0;
    int a[100];
    ifstream f ("GHH.inp");
    f>>n;
    for (int i=0;i<n;i++)
        f>>a[i];
    ofstream g ("GHH.out");
    for(int i=0;i<n;i++)
    if (check(a[i])==true)
    {
dem =dem +1;
    }
    g<<dem+1;

    f.close();
    g.close();
    return 0;
}
bool check (int n)
{
    int  sum =0;
    for (int i=1;i<=n/2;i++)
    {
     if (n%i==0)
      sum+=1;
    }
    if(sum==1) return true;
    return false;
}
